src/Executables/ReduceCceWorldtube/ReduceCceWorldtube.cpp

```diff
@@ -283,44 +391,60 @@ void perform_cce_worldtube_reduction(
-      "number of time steps to load during each call to the file-accessing "
-      "routines. Higher values mean fewer, larger loads from file into RAM.")(
-      "lmax_factor", boost::program_options::value<size_t>()->default_value(2),
-      "the boundary computations will be performed at a resolution that is "
-      "lmax_factor times the input file lmax to avoid aliasing");
+      "input-file", boost::program_options::value<std::string>()->required(),
+      "Name of YAML file to parse.");
 
   boost::program_options::variables_map vars;
-
   boost::program_options::store(
       boost::program_options::command_line_parser(argc, argv)
-          .positional(pos_desc)
           .options(desc)
           .run(),
       vars);
 
-  if (vars.count("help") != 0u or vars.count("input_file") == 0u or
-      vars.count("output_file") == 0u) {
-    Parallel::printf("%s\n", desc);
+  // Option parser for all the actual options
+  Options::Parser<option_tags> parser{
+      "This executable is used for converting the unnecessarily large SpEC "
+      "worldtube data format into a far smaller representation (roughly a "
+      "factor of 4) just storing the worldtube scalars that are required as "
+      "input for CCE."};
+
+  // Help is a successful return
+  if (vars.contains("help")) {
+    Parallel::printf("%s\n%s", desc, parser.help());
     return 0;
   }
 
-  perform_cce_worldtube_reduction(vars["input_file"].as<std::string>(),
-                                  vars["output_file"].as<std::string>(),
-                                  vars["buffer_depth"].as<size_t>(),
-                                  vars["lmax_factor"].as<size_t>(),
-                                  vars.count("fix_spec_normalization") != 0u);
+  // Not specifying an input file is an error
+  if (not vars.contains("input-file")) {
+    Parallel::printf("Missing input file. Pass '--input-file'");
+    return 1;
+  }
+
+  // Wrap in try-catch to print nice errors and terminate gracefully
+  try {
+    const std::string input_yaml = vars["input-file"].as<std::string>();
+
+    // Actually parse the yaml. This does a check if it exists.
+    parser.parse_file(input_yaml);
+
+    // First create option tags, and then actual tags.
+    const OptionTuple options = parser.template apply<option_tags>(
+        [](auto... args) { return OptionTuple(std::move(args)...); });
+    const TagsTuple inputs =
+        Parallel::create_from_options<void>(options, tags{});
+
+    // Do the reduction
+    perform_cce_worldtube_reduction(
+        tuples::get<ReduceCceTags::InputSpecH5File>(inputs),
+        tuples::get<ReduceCceTags::OutputH5File>(inputs),
+        tuples::get<ReduceCceTags::BufferDepth>(inputs),
+        tuples::get<ReduceCceTags::BufferDepth>(inputs),
+        tuples::get<ReduceCceTags::FixSpecNormalization>(inputs));
+  } catch (const std::exception& exception) {
+    Parallel::printf("%s\n", exception.what());
+    return 1;
```

Comment:
`return 2;`

Reply:
We actually use `2` in our charm execs to mean "continue from checkpoint". `src/Parallel/ExitCode.hpp` Should I still use that here?

Reply:
Oh, hmm, I was just thinking a number to distinguish from the other error. Maybe we should choose and document a number for "Input file parse failed" and then use `1` as " 🤷  it went bad" 

I'm happy to defer this until some other point. I think it's beyond the scope of this PR. Up to you!

Reply:
I like dedicating another number to "input file parse failed" and using `1` for general error. But yeah it's outside the scope of this PR, so I'll keep it as 1 for now and change it later.

Reply:
Sounds good!